### PR TITLE
Extract init scripts and state tracking

### PR DESCRIPTION
**Pull request by bigdaz**
This PR adds a few more improvements before enhancing the Job Summary output.

### CODE REVIEW

**Comment by bigdaz** [comment]
@eskatos Just FYI. I didn't end up requiring all of the `Path.join` methods and just used `Path.resolve(__dirname, '..', '..', 'src', 'resources', name)`

This seems to work on my test project but it would be great if you could confirm on `gradle/gradle`.

**Comment by eskatos** [comment]
It looks to work with `main`: https://github.com/gradle/gradle/actions/runs/2432914392

File: src/cache-base.ts [hunk 1]
```diff
@@ -169,72 +169,18 @@ export class GradleStateCache {
...
-        def projectRootEntry = settings.rootDir.absolutePath + "\\n"
-        def projectRootList = new File(settings.gradle.gradleUserHomeDir, "${PROJECT_ROOTS_FILE}")
-        if (!projectRootList.exists() || !projectRootList.text.contains(projectRootEntry)) {
-            projectRootList << projectRootEntry
-        }
-    }
-}`
-        )
+    private readResourceAsString(resource: string): string {
+        // Resolving relative to __dirname will force the compiler to inline the content in the distribution
```
Comment (eskatos) [inline src/cache-base.ts:181]: Looking into the _distribution_ files, the content of the init script isn't aligned.
The line below just resolve the path relative to the distribution. IOW at runtime `__dirname` is `./dist/main`.

Comment (bigdaz) [reply src/cache-base.ts:181]: Good point. I'll update the comment.

**Pull request merged**